_1_basics/src/main/java/code/_4_student_effort/PairOfTwo.java

```diff
@@ -0,0 +1,42 @@
+package code._4_student_effort;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class PairOfTwo {
+
+    int pairs (ArrayList<Integer> array) {
+        int number = 0;
+        int lastIndex = array.size() - 1;
+        int firstIndex = 0;
+        // sort the elements of ArrayList in ascending order
+        Collections.sort(array);
+
+        while (firstIndex < lastIndex) {

```

Comment:
ce complexitate ciclomatica are implementarea ta ? ma refer la big O notation ...

Reply:
Cred ca O(N*log(N))
Sortarea ar avea O(N*log(N)), iar while-ul (numararea perechilor) cred ca ar putea ajunge in cel mai rau caz la O(N)

Reply:
bravo